### ISSUE TITLE
Assess correctness of default YAML configs (especially normalization fallback)

### ISSUE THREAD

**Issue by NguyenJus** [NguyenJus/custom-sam-peft#69]
## Ask

Audit the shipped default YAML configs (`configs/examples/*.yaml`, `src/custom_sam_peft/cli/templates/*.yaml`) and confirm the defaults are the right *defaults* — not just values that happen to pass tests. Particular focus: **normalization (mean/std)**.

## Headline concern: normalization fallback may silently mis-normalize SAM3 inputs

All four user-facing configs ship the same normalize block:

```yaml
normalize:
  mean: [0.5, 0.5, 0.5]
  std:  [0.5, 0.5, 0.5]
```

(Files: `configs/examples/coco_text_{lora,qlora}.yaml`, `src/custom_sam_peft/cli/templates/coco_text_{lora,qlora}.yaml`. Schema default in `src/custom_sam_peft/config/schema.py` matches.)

`src/custom_sam_peft/data/transforms.py::resolve_normalization` is the gate:

1. Try `AutoImageProcessor.from_pretrained(model_name)`, read `image_mean` / `image_std`.
2. On `OSError | AttributeError | ValueError`, **fall back silently** to `NormalizeConfig.{mean,std}` and emit one INFO log line.

This means on the happy path the SAM3.1 processor's real stats win and the YAML values are irrelevant. But on the offline / cache-miss / wrong-revision / unsupported-model path, training proceeds with `[0.5, 0.5, 0.5]` — a generic [-1, 1] rescale — feeding a backbone pretrained on different stats. SAM/SAM2-class processors use ImageNet stats (`mean=[0.485, 0.456, 0.406]`, `std=[0.229, 0.224, 0.225]`). If SAM3.1 does too, the fallback is wrong in a way the user won't notice until eval mAP looks off.

### Questions to answer

- What does `Sam3ImageProcessor` / `AutoImageProcessor.from_pretrained("facebook/sam3.1")` actually return for `image_mean` / `image_std`?
- Should the fallback default match those exact stats (so the offline path is correct-by-default), or should we keep `[0.5, 0.5, 0.5]` as a deliberate "safe sentinel"?
- Should the fallback be an INFO log or a **WARN** — silent quality regressions are exactly the class of bug WARN exists for.
- Should we add a startup assertion comparing resolved `(mean, std)` against the model's expected stats and erroring (or loudly warning) on divergence beyond a tolerance?
- Should the YAML examples even *carry* a `normalize:` block, given it's a fallback the user usually shouldn't touch? Removing it from templates would prevent users from accidentally pinning bad stats.

## Other defaults worth a sanity pass while we're here

Not necessarily wrong — but each should be justified or adjusted:

- `data.image_size: 1008` — is this the model's native input size for SAM3.1, or a guess? Mismatched size silently triggers internal resize.
- `data.augmentations.color_jitter: 0.1` — modest; fine for segmentation but worth confirming for `prompt_mode: text` (CLIP-style text encoders can be jitter-sensitive).
- `data.text_prompt.negatives_per_image: 4` — arbitrary-looking; should be tied to expected class count / a documented heuristic.
- `train.lr: 1.0e-4`, `warmup_steps: 100`, `epochs: 10`, `grad_accum_steps: 8` — confirm these are sensible *defaults* (vs example-specific). LR in particular is suspect for QLoRA where the standard recipe is often higher (1e-3 ~ 5e-4 with `paged_adamw_8bit`).
- `train.box_hint.{p_start=1.0, p_end=0.0, decay_steps=5000, early_stop_p_threshold=0.05}` — schedule was chosen ad-hoc; revisit once #24 lands.
- `train.optimizer: auto` — confirm the resolution rule (LoRA → AdamW, QLoRA → paged 8-bit?) is documented and matches the spec.
- `eval.iou_thresholds: [0.5, ..., 0.95]` — COCO-style; confirm this is the right default outside COCO too.
- `model.gradient_checkpointing: false` with the inline reference to #60 — fine while that issue is open, but flag as a TODO to flip once #60 is resolved.

## Deliverables

1. Empirical: dump `AutoImageProcessor` stats for `facebook/sam3.1` and confirm/refute the `[0.5, 0.5, 0.5]` fallback.
2. For each default above: keep / change / remove + one-line justification.
3. If we change the fallback or the warn level, also update `src/custom_sam_peft/config/schema.py` defaults so the schema, examples, and CLI templates stay in lockstep (single source of truth for the shipped defaults).
4. If we add a divergence assertion, decide WARN vs hard error and document the override path.

## Out of scope

- Wholesale rework of the augmentation pipeline.
- Adding new config knobs — assessment is about whether current defaults are right, not about expanding surface area.
